tools/src/links.rs

```diff
@@ -12,11 +12,21 @@ pub fn linkify(s: String) -> String {
         last = link.end();
     }
     if last == 0 {
-        s
+        result = s;
     } else {
         result.push_str(&s[last ..]);
-        result
     }
+
+    linkify_bug_numbers(&result)
+}
+
+pub fn linkify_bug_numbers(s: &str) -> String {
+    lazy_static! {
+        static ref BUG_NUMBER_REGEX: Regex = {
+            Regex::new(r"\b(?i)bug\s*(?P<bugno>[1-9][0-9]{2,6})\b").unwrap()
```

Comment:
A comment here about why the regex is not just the number would be nice I think. We can always make the regex smarter in the future if we find uses with different patterns.

Reply:
Or alternatively, we could just call `linkify_bug_numbers` during the same linkify phase, like:

```diff
diff --git a/tools/src/links.rs b/tools/src/links.rs
index 3063cef..7d66410 100644
--- a/tools/src/links.rs
+++ b/tools/src/links.rs
@@ -7,14 +7,14 @@ pub fn linkify(s: String) -> String {
     let mut last = 0;
     let mut result = String::new();
     for link in finder.links(&s) {
-        result.push_str(&s[last .. link.start()]);
+        result.push_str(&linkify_bug_numbers(&s[last .. link.start()]));
         result.push_str(&format!("<a href=\"{}\">{}</a>", link.as_str(), link.as_str()));
         last = link.end();
     }
     if last == 0 {
-        s
+        linkify_bug_number(&s)
     } else {
-        result.push_str(&s[last ..]);
+        result.push_str(&linkify_bug_numbers(&s[last .. link.start()]));
         result
     }
 }
```

And keep the number replacement as-is.

Reply:
I tried something similar to that but it turns out a lot of comments have numbers that are not bug number references, so just looking for a number gets a lot of false positives. I'll add the comment like you suggested and we can look into adjusting the regex more in the future if we need to.